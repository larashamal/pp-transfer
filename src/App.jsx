import { useState } from "react";
import "./App.css";

function App() {
  const [balance, setBalance] = useState(100);
  const [amount, setAmount] = useState("");
  const [message, setMessage] = useState("");

  function handleClick() {
    if (amount <= balance && amount > 0) {
      setBalance(balance - amount);
      setMessage(`${amount} transferred.`);
    } else {
      setMessage(`insufficient funds`);
    }
  }

  const handleChange = (event) => setAmount(event.target.value);

  return (
    <>
      <p aria-label="balance">Â£{balance}</p>
      <input className="border" aria-label="input" onChange={handleChange} />
      <button type="button" onClick={handleClick}>
        {" "}
        Submit{" "}
      </button>

      <p>{message}</p>
    </>
  );
}

export default App;
