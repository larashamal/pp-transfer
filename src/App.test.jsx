import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import App from "./App";

it("renders without crashing", () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("updates balance when we enter a valid input", () => {
  render(<App />);
  // Arrange
  const balance = screen.getByLabelText(/balance/i);
  const input = screen.getByLabelText(/input/i);
  const button = screen.getByRole("button");

  // Act
  userEvent.type(input, "30");
  userEvent.click(button);

  // Assert
  expect(balance.textContent).toBe("Â£70");
});
