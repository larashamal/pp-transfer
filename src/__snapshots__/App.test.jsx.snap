// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <p
    aria-label="balance"
  >
    Â£
    100
  </p>,
  <input
    aria-label="input"
    className="border"
    onChange={[Function]}
  />,
  <button
    onClick={[Function]}
    type="button"
  >
     
    Submit
     
  </button>,
  <p />,
]
`;
